keel-web/src/main/kotlin/com/netflix/spinnaker/keel/services/ApplicationService.kt

```diff
@@ -142,6 +157,39 @@ class ApplicationService(
     )
   }
 
+  /**
+   * Adds details about any stateless constraints in the given environment to the [ArtifactSummaryInEnvironment].
+   */
+  private fun addStatelessConstraintSummaries(
+    artifactSummaryInEnvironment: ArtifactSummaryInEnvironment,
+    deliveryConfig: DeliveryConfig,
+    environment: Environment,
+    version: String,
+    artifact: DeliveryArtifact
+  ): ArtifactSummaryInEnvironment {
+    val statelessConstraints: List<StatelessConstraintSummary?> = environment.constraints.filter { constraint ->
+      constraint !is StatefulConstraint
+    }.map { constraint ->
```

Comment:
You can use `.mapNotNull` here, then you won't have any nulls in your list above.

Reply:
Tried changing it and it still didn't help with reducing the `null` checks..